ledger/index.ts

```diff
@@ -336,65 +334,182 @@ export async function* signLedgerMixedBtcTransaction(
 }
 
 /**
- * This function is used to sign an incoming Native Segwit Psbt with the ledger
+ * This function is used to sign an incoming Native Segwit / Taproot PSBT with the ledger
  * @param transport - the transport object with connected ledger device
  * @param network - the network type (Mainnet or Testnet)
  * @param addressIndex - the index of the account address to sign with
- * @param indexesToSign - the indexes of the inputs to sign
- * @param base64Psbt - the incoming transaction in base64 format
- * @returns the signed raw transaction in hex format
+ * @param inputToSign - the input with address and indexes to sign
+ * @param psbtBase64 - the incoming transaction in base64 format
+ * @param finalize - boolean to finalize the transaction
+ * @returns the signed PSBT in string (base64) format
  * */
-export async function signIncomingSingleSigNativeSegwitTransactionRequest(
+export async function signIncomingSingleSigPSBT(
   transport: Transport,
   network: NetworkType,
   addressIndex: number,
-  indexesToSign: number[],
-  base64Psbt: string
+  inputToSign: InputToSign,
+  psbtBase64: string,
+  finalize = false,
 ): Promise<string> {
+  if (!psbtBase64?.length) {
+    throw new Error('Invalid transaction');
+  }
+
   const coinType = getCoinType(network);
   const app = new AppClient(transport);
+  const { type } = getAddressInfo(inputToSign.address);
+  const isSegwit = type === 'p2wpkh';
 
   // Get account details from ledger to not rely on state
   const masterFingerPrint = await app.getMasterFingerprint();
   const extendedPublicKey = await app.getExtendedPubkey(`m/84'/${coinType}'/0'`);
   const accountPolicy = new DefaultWalletPolicy(
     'wpkh(@0/**)',
-    `[${masterFingerPrint}/84'/${coinType}'/0']${extendedPublicKey}`
+    `[${masterFingerPrint}/84'/${coinType}'/0']${extendedPublicKey}`,
   );
 
-  const { publicKey: senderPublicKey } = getNativeSegwitAccountDataFromXpub(
-    extendedPublicKey,
-    addressIndex,
-    network
-  );
+  const { publicKey: senderPublicKey } = getNativeSegwitAccountDataFromXpub(extendedPublicKey, addressIndex, network);
 
   const inputDerivation: Bip32Derivation = {
     path: `m/84'/${coinType}'/0'/0/${addressIndex}`,
     pubkey: senderPublicKey,
     masterFingerprint: Buffer.from(masterFingerPrint, 'hex'),
   };
 
-  const psbt = Psbt.fromBase64(base64Psbt);
+  const taprootExtendedPublicKey = await app.getExtendedPubkey(`m/86'/${coinType}'/0'`);
```

Comment:
we should probably extract this code in a util function and only call the function if the `isSegwit` flag is false and we can do the same for segwit

Reply:
I added this to the ledger refactoring task https://linear.app/xverseapp/issue/ENG-2537/refactor-ledger-related-logic if you don't mind

Reply:
that's ok